sphinx/directives/code.py

```diff
@@ -239,7 +239,7 @@ def read(self, location: Optional[Tuple[str, int]] = None) -> Tuple[str, int]:
 
     def show_diff(self, location: Optional[Tuple[str, int]] = None) -> List[str]:
         new_lines = self.read_file(self.filename)
-        old_filename = self.options.get('diff')
+        old_filename = self.options['diff']
```

Comment:
doesn't look to me like the `None` case was handled before, so i don't think this is any less safe than before